.github/actions/observe-build-status/action.yml

```diff
@@ -0,0 +1,60 @@
+# This action expects the code to have been checked out beforehand, e.g. via actions/checkout@v4
+#
+# This action expects certain secrets to be provided:
+#   - VAULT_ADDR
+#   - VAULT_ROLE_ID
+#   - VAULT_SECRET_ID
+---
+name: Observe Build Status
+description: Records the build status remotely for analytic purposes
+
+inputs:
+  build_status:
+    description: 'The status of the job, one of: success, failure, aborted, cancelled'
+    required: true
+  user_reason:
+    description: 'Optional string (200 chars max) the user can submit to indicate the reason why a build has ended with a certain build status.'
+    required: false
+  user_description:
+    description: 'Optional string (200 chars max) the user can submit to indicate the reason why a build has ended with a certain build status.'
+    required: false
+  job_name:
+    description: 'Optional string, the job whose status is being observed; defaults to $GITHUB_JOB when omitted'
+    required: false
+  secret_vault_address:
+    description: 'Secret vault url'
+    required: false
+  secret_vault_roleId:
+    description: 'Secret vault roleId'
+    required: false
+  secret_vault_secretId:
+    description: 'Secret vault ID'
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Import Secrets
+      id: secrets
+      uses: hashicorp/vault-action@v3.0.0
+      if: |
+        inputs.secret_vault_address != ''
+        && inputs.secret_vault_roleId != ''
+        && inputs.secret_vault_secretId != ''
+      with:
+        url: ${{ inputs.secret_vault_address }}
+        method: approle
+        roleId: ${{ inputs.secret_vault_roleId }}
+        secretId: ${{ inputs.secret_vault_secretId }}
+        exportEnv: false # we rely on step outputs, no need for environment variables
+        secrets: |
+          secret/data/products/zeebe/ci/ci-analytics gcloud_sa_key;
+    - uses: camunda/infra-global-github-actions/submit-build-status@main
+      if: ${{ always() && steps.secrets.outputs.gcloud_sa_key != '' }}
+      with:
+        job_name_override: "${{ inputs.job_name }}"
+        build_status: "${{ inputs.build_status || (contains(steps.*.outcome, 'failure') && 'failure' || 'success') }}"
```

Comment:
:question: Does this work? I could imagine that `steps` are only the steps in this composite action, not the steps of the job this action is being called from.

Reply:
Oh, right, good point. I'll double check. If it doesn't, we can just pass it as input.